src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,126 @@
+import games.strategy.engine.delegate.AutoSave;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.engine.message.IRemote;
+import games.strategy.engine.random.IRandomStats.DiceType;
+import games.strategy.triplea.MapSupport;
+import games.strategy.triplea.TripleAUnit;
+import games.strategy.triplea.attachments.PlayerAttachment;
+import games.strategy.triplea.attachments.TerritoryAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.IBattle.BattleType;
+import games.strategy.triplea.delegate.IBattle.WhoWon;
+import games.strategy.triplea.delegate.dataObjects.BattleListing;
+import games.strategy.triplea.delegate.dataObjects.BattleRecord;
+import games.strategy.triplea.delegate.remote.IBattleDelegate;
+import games.strategy.triplea.formatter.MyFormatter;
+import games.strategy.triplea.oddsCalculator.ta.BattleResults;
+import games.strategy.triplea.player.ITripleAPlayer;
+import games.strategy.util.CompositeMatch;
+import games.strategy.util.CompositeMatchAnd;
+import games.strategy.util.CompositeMatchOr;
+import games.strategy.util.IntegerMap;
+import games.strategy.util.Match;
+import games.strategy.util.Tuple;
+
+public class BattleDelegateOrdered extends BattleDelegate {
+
+  /**
+   * Called before the delegate will run.
+   */
+  @Override
+  public void start() {
+    super.start();
+    // we may start multiple times due to loading after saving
+    // only initialize once
+
+    
+    IBattle battle;
+    for( final Territory t : m_battleTracker.getPendingBattleSites(true) ) {   // loop throught bombing/air raids
+      battle = m_battleTracker.getPendingBattle(t, true, null);
+      if( battle == null ) {
+        try {
+          throw new Exception("Air/Bombing Raid gone missing in BattleDelegate");
+        } catch( Exception e ) {  // Will crash below
+        }
+      }
+      battle.fight(m_bridge);
+      battle = m_battleTracker.getPendingBattle(t, true, BattleType.BOMBING_RAID );  // check to see if there's still a bombing raid for the territory - i.e. previous battle was an air raid
+      if( battle != null ) {
+        battle.fight(m_bridge);
+      }
+    }
+
+    int otherBattleCount = 0;
+    int amphibCount = 0;
+    IBattle lastAmphib = null;
+    for( final Territory t : m_battleTracker.getPendingBattleSites(false) ) {  // Loop through normal combats i.e. not bombing or air raid
+      battle = m_battleTracker.getPendingBattle(t, false, BattleType.NORMAL);
+      // we only care about battles where we must fight
+      // this check is really to avoid implementing getAttackingFrom() in other battle subclasses
+      if (!(battle instanceof MustFightBattle) ) {
+        continue;
+      }
+      if( !t.isWater() ) {
+        final Map<Territory, Collection<Unit>> attackingFromMap = ((MustFightBattle) battle).getAttackingFromMap();
+        for( final Territory neighbor : ((MustFightBattle) battle).getAttackingFrom() ) {
```

Comment:
If you think it is a naff design, why are you going with it? This is a lot of green, new code..